### PR TITLE
SDK functions for auditus

### PR DESCRIPTION
**Pull request by br41nslug**
## Scope

What's changed:

- Implemented new SDK functions for Auditus
- Updated WebSocket subscriptions to include the new Auditus collections
- Implemented new GraphQL queries for Auditus

## Potential Risks / Drawbacks

- Lorem ipsum dolor sit amet
- Consectetur adipiscing elit

## Review Notes / Questions

- 

## TODO

- [x] SDK functions
- [x] Update websocket subscriptions
- [x] Add new GQL endpoints

### CODE REVIEW

File: .changeset/nine-geckos-jog.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@directus/api": minor
+---
+
+Updated WebSocket subscriptions to include the new policies
```
Comment (paescuj) [inline .changeset/nine-geckos-jog.md:5]:
```suggestion
Updated WebSocket subscriptions to include the new policies collection
```

File: api/src/services/graphql/index.ts [hunk 2]
```diff
@@ -3232,6 +3244,81 @@ export class GraphQLService {
 			});
 		}
 
+		if ('directus_permissions' in schema.read.collections) {
+			schemaComposer.Query.addFields({
+				permissions_me: {
+					type: schemaComposer.createScalarTC<CollectionAccess>({
+						name: 'test',
```
Comment (paescuj) [inline api/src/services/graphql/index.ts:3251]: Should this be something like
```suggestion
						name: 'permissions_me_type',
```
?

Comment (br41nslug) [reply api/src/services/graphql/index.ts]: Oops forgot to rename the test type 🙈

File: sdk/src/rest/commands/create/policies.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	Item extends object = DirectusPolicy<Schema>,
+> = ApplyQueryFields<Schema, Item, TQuery['fields']>;
+
+/**
+ * Create multiple new policies
+ *
+ * @param items The policy to create
+ * @param query Optional return data query
+ *
+ * @returns Returns the policy objects for the created policy.
```
Comment (paescuj) [inline sdk/src/rest/commands/create/policies.ts:17]:
```suggestion
 * @returns Returns the policy objects for the created policies.
```

Comment (paescuj) [inline sdk/src/rest/commands/create/policies.ts:17]:
```suggestion
 * @param items The policies to create
 * @param query Optional return data query
 *
 * @returns Returns the policy objects for the created policies.
```

File: sdk/src/rest/commands/create/policies.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		method: 'POST',
+	});
+
+/**
+ * Create a new policy
+ *
+ * @param item The policy to create
+ * @param query Optional return data query
+ *
+ * @returns Returns the policy for the created policy.
```
Comment (paescuj) [inline sdk/src/rest/commands/create/policies.ts:37]:
```suggestion
 * @returns Returns the policy object for the created policy.
```

File: sdk/src/rest/commands/read/policies.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+> = ApplyQueryFields<Schema, Item, TQuery['fields']>;
+
+export type ReadPolicyGlobalsOutput = {
+	app_access: boolean;
+	admin_access: boolean;
+	enforce_tfa: boolean;
+};
+
+/**
+ * List all policies that exist in Directus.
```
Comment (paescuj) [inline sdk/src/rest/commands/read/policies.ts:19]:
```suggestion
 * List all policies that exist in the project.
```

File: sdk/src/rest/commands/read/policies.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+ * @param query The query parameters
+ * @returns An array of up to limit Permission objects. If no items are available, data will be an empty array.
```
Comment (paescuj) [inline sdk/src/rest/commands/read/policies.ts:21]:
```suggestion
 * @returns An array of up to limit Policy objects. If no items are available, data will be an empty array.
```

File: sdk/src/rest/commands/read/policies.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+		path: `/policies`,
+		params: query ?? {},
+		method: 'GET',
+	});
+
+/**
+ * Read a specific policy.
+ * @param key The primary key of the permission
+ * @param query The query parameters
+ * @returns Returns a Permission object if a valid primary key was provided.
```
Comment (paescuj) [inline sdk/src/rest/commands/read/policies.ts:37]:
```suggestion
 * @returns Returns a Policy object if a valid primary key was provided.
```

File: sdk/src/rest/commands/read/policies.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+			params: query ?? {},
+			method: 'GET',
+		};
+	};
+
+/**
+ * Check the current user's global policy.
+ * @param collection The collection of the item
+ * @param key The primary key of the item
+ * @returns Return
```
Comment (paescuj) [inline sdk/src/rest/commands/read/policies.ts:59]:
```suggestion
 * Check the current user's policy globals.
```

File: sdk/src/rest/commands/read/roles.ts [hunk 9]
```diff
@@ -45,3 +45,19 @@ export const readRole =
 			method: 'GET',
 		};
 	};
+
+/**
+ * List the attached roles for the current user.
+ * @param query The query parameters
+ * @returns Returns a Role object if a valid primary key was provided.
```
Comment (paescuj) [inline sdk/src/rest/commands/read/roles.ts:52]:
```suggestion
 * @returns Returns Role objects
```

File: sdk/src/schema/policy.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import type { MergeCoreCollection } from '../index.js';
+import type { DirectusPermission } from './permission.js';
+import type { DirectusRole } from './role.js';
+import type { DirectusUser } from './user.js';
+
+/**
+ * directus_policies type
+ */
```
Comment (paescuj) [inline sdk/src/schema/policy.ts:8]: Since the other types don't have such a comment either
```suggestion
```

File: sdk/src/schema/policy.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+export type DirectusPolicy<Schema> = MergeCoreCollection<
+	Schema,
+	'directus_policies',
+	{
+		id: string; // uuid
+		name: string;
```
Comment (paescuj) [inline sdk/src/schema/policy.ts:11]: Has now also an icon 🎨 
```suggestion
		icon: string | null;
```

Comment (hanneskuettner) [reply sdk/src/schema/policy.ts]: Nitpick, the icon is always string as the column is non nullable

**Review by paescuj — APPROVED** [review]
Great work 🎈

**Pull request merged**